text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1601 @@
+message Sample {
+  // The indices recorded here correspond to locations in Profile.location.
+  // The leaf is at location_index[0]. [deprecated, superseded by locations_start_index / locations_length]
+  repeated uint64 location_index = 1;
+  // locations_start_index along with locations_length refers to to a slice of locations in Profile.location.
+  // Supersedes location_index.
+  uint64 locations_start_index = 7;
+  // locations_length along with locations_start_index refers to a slice of locations in Profile.location.
+  // Supersedes location_index.
+  uint64 locations_length = 8;
+  // A 128bit id that uniquely identifies this stacktrace, globally. Index into string table. [optional]
+  uint32 stacktrace_id_index = 9;
+  // The type and unit of each value is defined by the corresponding
+  // entry in Profile.sample_type. All samples must have the same
+  // number of values, the same as the length of Profile.sample_type.
+  // When aggregating multiple samples into a single sample, the
+  // result has a list of values that is the element-wise sum of the
+  // lists of the originals.
+  repeated int64 value = 2;
+  // label includes additional context for this sample. It can include
+  // things like a thread id, allocation size, etc.
+  //
+  // NOTE: While possible, having multiple values for the same label key is
+  // strongly discouraged and should never be used. Most tools (e.g. pprof) do
+  // not have good (or any) support for multi-value labels. And an even more
+  // discouraged case is having a string label and a numeric label of the same
+  // name on a sample.  Again, possible to express, but should not be used.
+  // [deprecated, superseded by attributes]
+  repeated Label label = 3;
+  // References to attributes in Profile.attribute_table. [optional]
+  repeated uint64 attributes = 10;
+
+  // Reference to link in Profile.link_table. [optional]
+  uint64 link = 12;
+
+  // Timestamps associated with Sample represented in ms. These timestamps are expected
+  // to fall within the Profile's time range. [optional]
+  repeated uint64 timestamps = 13;
+}
+
+// Provides additional context for a sample,
+// such as thread ID or allocation size, with optional units. [deprecated]
+message Label {
+  int64 key = 1;   // Index into string table
+
+  // At most one of the following must be present
+  int64 str = 2;   // Index into string table
+  int64 num = 3;
+
+  // Should only be present when num is present.
+  // Specifies the units of num.
+  // Use arbitrary string (for example, "requests") as a custom count unit.
+  // If no unit is specified, consumer may apply heuristic to deduce the unit.
+  // Consumers may also  interpret units like "bytes" and "kilobytes" as memory
+  // units and units like "seconds" and "nanoseconds" as time units,
+  // and apply appropriate unit conversions to these.
+  int64 num_unit = 4;  // Index into string table
+}
+
+// Indicates the semantics of the build_id field.
+enum BuildIdKind {
```

Comment:
For Windows, the equivalent of GNU build note are the various codeview sections in a PE file:
https://www.debuginfo.com/articles/debuginfomatch.html

I would suggest adding BUILD_ID_PDB_20 and BUILD_ID_PDB_70 as those are very commonly tracked values for Windows based profiling for symbol correlation.

The .NET team also added a newer convention for matching a Linux perfmap file in their PE files for ready-to-run compilation that should be covered here:
https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/r2r-perfmap-format.md
https://github.com/dotnet/runtime/pull/53792

I would suggest a BUILD_ID_R2R_PERF_MAP for this.

Reply:
I would suggest we submit this PR as is and then have subsequent change to add these other build ID types as enum values so that the discussion and evolution are clearer. You could send that PR, doesn't have to be Dmitry. We'll need to add proper comments to hint at what the semantics of each value are.

Reply:
Sounds good to me.

Reply:
No, this format does not specifically deal with JIT.